### PR TITLE
Better exception message when sort on sub object

### LINKED ISSUE #2034: Exception message of `RealmQuery.findAllSorted` when fieldName contains `.`
`Field name '%s' does not exist.` is not a clear enough. Should be the same with `RealmResults.sort` "Sorting using child object properties is not supported: "

### PR DESCRIPTION
**Pull request by beeender**
Close #2034

### CODE REVIEW

**Comment by cmelchior** [comment]
retest this please

**Comment by dalinaum** [comment]
:+1:

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 1]
```diff
@@ -1934,6 +1930,21 @@ private long getSourceRowIndexForFirstObject() {
         }
     }
 
+    // Get the column index for sorting related functions. A proper exception will be thrown if the field doesn't exist
+    // or it belongs to the child object.
+    private long getColumnIndexForSort(String fieldName) {
+        if (fieldName.contains(".")) {
+            throw new IllegalArgumentException("Sorting using child object properties is not supported: " + fieldName);
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java]: properties -> fields

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 2]
```diff
@@ -1610,6 +1607,8 @@ private boolean isDynamicQuery() {
      * @return immediately an empty {@link RealmResults}. Users need to register a listener
      * {@link io.realm.RealmResults#addChangeListener(RealmChangeListener)} to be notified when the query completes.
      * @see io.realm.RealmResults
+     * @throws java.lang.IllegalArgumentException if one of the field names does not exist or it belongs to a sub
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java]: sub -> "referenced" or "child"

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 3]
```diff
@@ -1564,7 +1563,8 @@ public Long call() throws Exception {
      * @param sortOrders how to sort the field names.
      * @return a {@link io.realm.RealmResults} containing objects. If no objects match the condition, a list with zero 
      *         objects is returned.
-     * @throws java.lang.IllegalArgumentException if a field name does not exist.
+     * @throws java.lang.IllegalArgumentException if one of the field names does not exist or it belongs to a sub
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java]: sub -> "referenced" or "child"

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 4]
```diff
@@ -1450,14 +1448,12 @@ public Long call() throws Exception {
      *
      * @return immediately an empty {@link RealmResults}. Users need to register a listener
      *         {@link io.realm.RealmResults#addChangeListener(RealmChangeListener)} to be notified when the query completes.
-     * @throws java.lang.IllegalArgumentException if field name does not exist.
+     * @throws java.lang.IllegalArgumentException if field name does not exist or it belongs to a sub
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java]: sub -> "referenced" or "child"

File: realm/realm-library/src/main/java/io/realm/RealmQuery.java [hunk 5]
```diff
@@ -1420,16 +1420,14 @@ public Long call() throws Exception {
      * @param sortOrder how to sort the results.
      * @return a {@link io.realm.RealmResults} containing objects. If no objects match the condition, a list with zero
      * objects is returned.
-     * @throws java.lang.IllegalArgumentException if field name does not exist.
+     * @throws java.lang.IllegalArgumentException if field name does not exist or it belongs to a sub
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/RealmQuery.java]: sub -> "referenced" or "child"

**Comment by cmelchior** [comment]
Apart from some small javadoc changes needed, this looks good :+1:

**Comment by beeender** [comment]
retest this please

**Pull request merged**